types/index.d.ts

```diff
@@ -1350,6 +1350,9 @@ declare namespace RamdaAdjunct {
          */
         sortByProps(props: string[], list: object[]): object[];
         sortByProps(props: string[]): (list: object[]) => object[];
+
+        /** Determine if input value is an indexed data type */
+        isIndexed(val: any): boolean;
```

Comment:
I think typescript allows following syntax
```suggestion
        isIndexed(val: any): val is string | any[];
```